aksparse/client/aksparse/sparse.py

```diff
@@ -127,48 +124,21 @@ def mul_vector(self, vector):
         return
 
 
-class coo(sparse_matrix):
+class Coo(SparseMatrix):
     """ Coordinate Format (COO) """
 
-    def __init__(self, data=None, columns=None, rows=None, shape=None, dense_matrix=None, no_chpl=None, from_csr=None, from_csc=None):
+    def __init__(self, data=None, columns=None, rows=None, shape=None, no_chpl=None, from_csr=None, from_csc=None):
         self.data = None
         self.columns = None
         self.rows = None
         self.shape = shape
 
-        # Build from existing matrix
-        if dense_matrix:
-            v, x, y = [], [], []
-            for row in range(len(dense_matrix)):
-                for col in range(len(dense_matrix[0])):
-                    if dense_matrix[row][col] != 0:
-                        v.append(dense_matrix[row][col])
-                        x.append(col)
-                        y.append(row)
-            resp = cast(str, ak.client.generic_msg(cmd="coo_construct", args={"data": v, "columns": x, "rows": y}))
-            arr_ids = resp.split("+")
-            self.data = ak.create_pdarray(f"created {arr_ids[0]}")
-            self.columns = ak.create_pdarray(f"created {arr_ids[1]}")
-            self.rows = ak.create_pdarray(f"created {arr_ids[2]}")
-            if shape == None:
-                self.shape = (len(dense_matrix[0]), len(dense_matrix))
-            else:
-                self.shape = shape
-        elif from_csr:
-            self.data = data
-            self.rows = rows
-            self.columns = columns
-            self.shape = shape
-        elif from_csc:
-            self.data = data
-            self.rows = rows
-            self.columns = columns
-            self.shape = shape
-        elif no_chpl:
-            self.data = data
+        if from_csr or from_csc or no_chpl:
             self.rows = rows
             self.columns = columns
             self.shape = shape
+            sorting_gb = ak.GroupBy([self.rows, self.columns])
+            self.data = data[sorting_gb.permutation]
```

Comment:
i wonder if a `coargsort` wouldn't be better if all we need is to sort the data
```python
self.data = data[ak.coargsort([self.rows, self.columns])]
```
Note: I have tested out the above. So I would def verify it gives what you would expect first